shopfloor/models/stock_move.py

```diff
@@ -4,12 +4,17 @@
 class StockMove(models.Model):
     _inherit = "stock.move"
 
-    def split_other_move_lines(self, move_lines):
+    def split_other_move_lines(self, move_lines, intersection=False):
         """Substract `move_lines` from `move.move_line_ids`, put the result
         in a new move and returns it.
+
+        If `intersection` is set to `True`, this is the common lines between
+        `move_lines` and `move.move_line_ids` which will be put in a new move.
         """
         self.ensure_one()
         other_move_lines = self.move_line_ids - move_lines
+        if intersection:
+            other_move_lines = self.move_line_ids & move_lines
```

Comment:
```suggestion
        if intersection:
            other_move_lines = self.move_line_ids & move_lines
        else:
            other_move_lines = self.move_line_ids - move_lines
```
to avoid interpolating recordsets 2 times

Reply:
Fixed